app/misc/graphing.ts

```diff
@@ -10,15 +10,36 @@ let commitHistory = [];
 let commitList = [];
 let spacingY = 100;
 let spacingX = 80;
+let tagSpacingY = 100;
+let tagSpacingX = 300;
 let parentCount = {};
 let columns: boolean[] = [];
 let edgeDic = {};
 let numOfCommits = 0;
 let branchIds = {};
+let tagIds = {};
+let unumberPrev = 0;
+
+// In order to allow tags, branches, and nodes to have unique numerical id's
+// reference: https://stackoverflow.com/questions/8012002/create-a-unique-number-with-javascript-time
+function generateUniqueNumber() {
+    var date = Date.now();
+
+    // If created at same millisecond as previous
+    if (date <= unumberPrev) {
+        date = ++unumberPrev;
+    } else {
+        unumberPrev = date;
+    }
+
+    return date;
+}
+
 
 function processGraph(commits: nodegit.Commit[]) {
     var promise = new Promise(function(resolve,reject){
```

Comment:
Peanut Gallery: What a bizarrely name Class. (I know you didn't create this, but the name perplexes me).

Reply:
_ALL_ of the classes are badly named to be honest. They don't entirely make sense. Thus my initiative to write a wiki explaining what's going on.